samples/interactivePersonalAccount.ts

```diff
@@ -0,0 +1,74 @@
+ *
+ * The workaround is to pass a known tenant Id to the `domain` property in the optional parameters when using these authentication methods. You can get the tenant Id from Azure portal or the Azure CLI.
+ *
+ * If you need to fetch the tenant Id programmatically:
+ *
+ * - Use any of the authentication methods without setting the domain to get a credential.
+ * - Call the `buildTenantLists(credential)` method by sending that same credential as the first parameter to get the list of all tenants in your account.
+ *
+ * If you want to avoid having to create a new credential, once you have access to one of the account's tenants, you can set it as the `domain` of your credentials by using the method `setDomain(tenant)` from the existing credentials object.
+ *
+ * This sample shows when the authentication fails for personal accounts, and how to overcome this issue.
+ *
+ * For context: https://github.com/Azure/ms-rest-nodeauth/issues/89#issuecomment-643471343
+ */
+import * as msRestNodeAuth from "../lib/msRestNodeAuth";
+import { SubscriptionClient } from "@azure/arm-subscriptions";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+// Copy the content of "sample.env" to a file named ".env". It should be stored at the root of the repo.
+// then from the root of the cloned repo run, ts-node ./samples/interactive.ts
+
+async function main(): Promise<void> {
+  const authentication = await msRestNodeAuth.interactiveLoginWithAuthResponse();
+
+  // For personal accounts, the below will print an empty array as we did not set the domain.
+  console.log(
+    "Subscriptions retrieved by default",
+    authentication.subscriptions
+  );
+
+  const client = new SubscriptionClient(authentication.credentials);
+
+  // Note: You must replace `<my-subscription>` with the Id of one of your subscriptions to get the error.
+  // If you use this code as it is, you will get `The subscription '<my-subscription>' could not be found.`,
+  // which is not an error relevant to the issue we're describing.
+  const subscriptionId = "<my-subscription>";
+
+  // Some requests will fail for personal accounts until we update the domain on the credential.
+  try {
+    await client.subscriptions.get(subscriptionId);
+  } catch (e) {
+    console.log(`
+Expected error:
+For personal accounts, we won't be able to retrieve subscriptions unless we specify a domain in the credentials.
+${e}
+`);
+  }
+
+  // To get the tenants for your account, you can use the buildTenantList method:
+  const tenants = await msRestNodeAuth.buildTenantList(
+    authentication.credentials
+  );
+
+  // If you have already authenticated and you want to retrieve the subscriptions of a specific tenant,
+  // you can set the credential's domain to the tenant you want to use,
+  // then retrieve your tenant's subscriptions with the SubscriptionsClient from @azure/arm-subscriptions:
```

Comment:
By this point, the user has already authenticated, so we shouldnt need to talk about "If you have already authenticated..."

```suggestion
  // Update the domain used by the credentials so that it can generate token against the right tenant
```